File: packages/examples/basic_dataclass/python-modules/basic_dataclass.py
from dataclasses import dataclass

@dataclass
class Person:
    name: str
    x: int = 0
    y: int = 0

    def move(self, dx: int, dy: int) -> None:
        self.x += dx
        self.y += dy

    def __str__(self) -> str:
        return f"{self.name} @ ({self.x}, {self.y})"

File: packages/examples/basic_dataclass/lib/basic_dataclass.dart
import "package:python_ffi_dart/python_ffi_dart.dart";

export "python_modules/basic_dataclass.g.dart";

final class Person extends PythonClass {
  factory Person(String name) => PythonFfiDart.instance.importClass(
        "basic_dataclass",
        "Person",
        Person.from,
        <Object?>[name],
      );

  Person.from(super.pythonClass) : super.from();

  void move(int dx, int dy) => getFunction("move").call(<Object?>[dx, dy]);
}
